pkg/reconciler/broker/trigger/controller.go

```diff
@@ -69,7 +69,21 @@ func NewController(
 	r.sourceTracker = duck.NewListableTrackerFromTracker(ctx, source.Get, impl.Tracker)
 	r.uriResolver = resolver.NewURIResolverFromTracker(ctx, impl.Tracker)
 
-	triggerInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
+	triggerInformer.Informer().AddEventHandler(controller.HandleAll(
+		func(obj interface{}) {
+			if trigger, ok := obj.(*eventingv1.Trigger); ok {
+				broker, err := brokerInformer.Lister().Brokers(trigger.Namespace).Get(trigger.Spec.Broker)
```

Comment:
If you hoist this logic into a function akin to getTriggersForBroker (say isMyTrigger or something), then we could actually UT this like we do for the getTriggersForBroker.
Also, wondering if this could be:
triggerInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
  FilterFunc: isMyTrigger. // this would call the new func for filtering?
  Handler: controller.HandleAll(impl.Enqueue)
  })

Reply:
in the kafka broker we also have a `func` that checks for "its" triggers.

Reply:
In addition, we need the same filter logic on leader promotion https://github.com/knative-sandbox/eventing-kafka-broker/blob/c3ba96be7bd4de07c2d66c98f4a699578e798e3e/control-plane/pkg/reconciler/trigger/controller.go#L86

Reply:
Awesome, thanks! I added that integration. Likely this all could use a followup to move it to a utility method to be shared by all the down streams, but I will leave that to you fine folks.